docs/content/3.middleware/2.guides/10.working-with-logger.md

```diff
@@ -0,0 +1,311 @@
+# Working with logger
+
+The middleware application provides a logger instance that automatically attaches metadata related to the scope of each call. This scope includes crucial information such as the name of the integration, the extension (if applicable), the function name (if applicable), the hook name (if applicable), and the type of the action being logged (e.g., endpoint, bootstrapHook, requestHook). By adding this contextual data, the logger makes it significantly easier to trace the origin of logs or errors, simplifying the debugging and monitoring process across the application.
+
+## Using logger
+
+The middleware application provides access to the logger in various parts of the system, such as in extensions and integrations. This flexibility allows you to log important events, errors, and other data throughout the lifecycle of your application. Whether you’re working in an extension or an integration, you can easily utilize the logger to capture and track necessary information.
+
+In this section, we will explore how to access and use the logger in different parts of the application, ensuring you have the tools to log data efficiently and consistently. 
+
+These examples will demonstrate the proper ways to obtain and interact with the logger, ensuring that you are able to implement consistent and effective logging practices across your middleware application. As you will see, `getLogger` function solves the problem in most cases.
```

Comment:
Maybe we can add output examples, it would be nice to show how our logger improve contextual data. If we do not want to add such example in all cases (that might be too specific as in my first comment), maybe we can add at least one such example.